trackintel/preprocessing/trips.py

```diff
@@ -0,0 +1,425 @@
+    return tours_df
+
+
+def _check_same_loc(stp1, stp2, staypoints):
+    """Check whether two staypoints are at the same location
+
+    Parameters
+    ----------
+    stp1 : int
+        First staypoint id
+    stp2 : int
+        Second staypoint id
+    staypoints : Trackintel staypoints
+        GeoDataFrame with staypoints and also location ids
+
+    Returns
+    -------
+    share_location, bool
+        If True, stp1 and stp2 are at the same location
+    """
+    if pd.isna(stp1) or pd.isna(stp2):
+        return False
+    share_location = staypoints.loc[stp1, "location_id"] == staypoints.loc[stp2, "location_id"]
+    return share_location
+
+
+def _check_max_dist(p1, p2, max_dist, crs_is_projected=False):
+    """
+    Check whether two points p1, p2 are less or equal than max_dist apart
+
+    Parameters
+    --------
+    p1, p2: shapely Point objects
+    max_dist: int
+
+    Returns
+    ------
+    dist_below_thresh: bool
+        indicating whether p1 and p2 are less than max_dist apart
+    """
+    if crs_is_projected:
+        dist = p1.distance(p2)
+    else:
+        dist = ti.geogr.point_distances.haversine_dist(p1.x, p1.y, p2.x, p2.y)
+    dist_below_thresh = dist <= max_dist
+    return dist_below_thresh
+
+
+def _create_tour_from_stack(temp_tour_stack, staypoints, max_time):
+    """
+    Aggregate information of tour elements in a structured dictionary.
+
+    Parameters
+    ----------
+    temp_tour_stack : list
+        list of dictionary like elements (either pandas series or python dictionary).
+        Contains all trips that will be aggregated into a tour
+
+    Returns
+    -------
+    tour_dict_entry: dictionary
+
+    """
+    # this function return and empty dict if no tripleg is in the stack
+    first_trip = temp_tour_stack.iloc[0]
+    last_trip = temp_tour_stack.iloc[-1]
+
+    # get location ID if available:
+    if staypoints is not None:
+        start_loc = staypoints.loc[first_trip["origin_staypoint_id"], "location_id"]
+        # double check whether start and end location are the same
+        end_loc = staypoints.loc[last_trip["destination_staypoint_id"], "location_id"]
+        assert start_loc == end_loc
+    else:
+        # set location to NaN since not available
+        start_loc = pd.NA
+
+    # all data has to be from the same user
+    assert len(temp_tour_stack["user_id"].unique()) == 1
+
+    # double check if tour requirements are fulfilled
+    assert last_trip["finished_at"] - first_trip["started_at"] <= max_time
+
+    tour_dict_entry = {
+        "user_id": first_trip["user_id"],
+        "started_at": first_trip["started_at"],
+        "finished_at": last_trip["finished_at"],
+        "origin_staypoint_id": first_trip["origin_staypoint_id"],
+        "destination_staypoint_id": last_trip["destination_staypoint_id"],
+        "trips": list(temp_tour_stack.index),
+        "location_id": start_loc,
+    }
+
+    return tour_dict_entry
+
+
+def get_trips_grouped(trips, tours):
```

Comment:
I could not find the usage of this function in our code, so this is a function exposed to the users and maybe not a helper function for us? Then consider moving this function to the beginning of the file. 

Somewhile ago we though it might be good to have a standard for placing functions, [here](https://github.com/mie-lab/trackintel/blob/master/CONTRIBUTING.md#order-of-functions-in-code)

Reply:
Yes, it's a helper function I wrote because of the problem that we have a n-to-n relation ship between trips and tours (one trip can be part of several tours). I moved it to the top of the file now.